CAIPs/caip-22.md

```diff
@@ -0,0 +1,62 @@
+---
+caip: 22
+title: Asset Reference for the ERC721 Asset Namespace
+author: Antoine Herzog (@antoineherzog), Pedro Gomes (@pedrouid), Joel Thorstensson (@oed)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/pull/19
+status: Draft
+type: Standard
+created: 2020-06-23
+updated: 2020-06-23
+requires: 10
+---
+
+## Simple Summary
+
+This document is about the details of the ERC721 asset namespace and reference for CAIP-19.
+
+## Abstract
+
+In CAIP-19 a general asset identification scheme is defined. This is the
+implementation of CAIP-19 for ERC721 (referencing NFT tokens).
+
+## Motivation
+
+See CAIP-19.
+
+## Specification
+
+### ERC721 Asset Namespace
+
+The asset namespace is called "erc721" as in [ERC721](https://eips.ethereum.org/EIPS/eip-721). It reference erc721 tokens (NFT) in the eip155 namespace (cf. CAIP3).
+
+#### Asset Reference Definition
+
+The Asset Reference format is the smart contract address of the erc721 token in the current chain_id.
+
+#### Token ID Definition
+
+The Token Id format is the NFT Identifier of the erc721 specification in the current chain_id.
+
+## Rationale
+
+The smart contract address strives for uniqueness for any erc721 tokens in the chain_id scope.
+
+## Backwards Compatibility
+
+Not applicable
+
+## Test Cases
+
+This is a list of manually composed examples
+
+```
+# CryptoKitties Collectible
+eip155:1/erc721:0x06012c8cf97BEaD5deAe237070F9587f8E7A266d
```

Comment:
What's the purpose of having a way to refer to an ERC721 contract?

Reply:
Exchanges and Wallet usually needs to give a human readable name or abreviation  for different type of assets.

 It can be a ticker for fungible like
XBT for bitcoin -> bip122:000000000019d6689c085ae165831e93/slip44:0 

or giving a name for a type of collectible. It can be quite usefull for signed contract between 2 parties for example.

cryptokitties -> eip155:1/erc721:0x06012c8cf97BEaD5deAe237070F9587f8E7A266d

We could imagine a universal asset registry which would register asset type name.  Then no need for each exchange, each wallet to redo this painfull work all the time again.

Reply:
Makes sense, ty!